src/App.tsx

```diff
@@ -1,14 +1,89 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { FilterCases } from './types/FilterCases';
+
+function filterBySelect(
+  filter: FilterCases,
+  todos: Todo[],
+) {
+  switch (filter) {
+    case FilterCases.Active:
+      return todos
+        .filter(({ completed }) => !completed);
+
+    case FilterCases.Completed:
+      return todos
+        .filter(({ completed }) => completed);
+
+    default:
+      return todos;
+  }
+}
+
+function getTodo(todos: Todo[], id: number) {
+  return todos.find(todo => todo.id === id);
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectFilter, setSelectFilter] = useState<FilterCases>(FilterCases.All);
+  const [activeTodo, setActiveTodo] = useState<Todo | undefined>(undefined);
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+  const [isOpenModal, setIsOpenModal] = useState(false);
+
+  const handleQueryUpdate = (value: string) => {
+    setQuery(value);
+  };
+
+  const handleSelectUpdate = (value: FilterCases) => {
+    setSelectFilter(value);
+  };
+
+  const handleTodoClick = async (id?: number) => {
+    setIsOpenModal(current => !current);
+
+    if (id) {
+      const newActiveTodo = getTodo(todos, id);
+      const user = await getUser(newActiveTodo?.userId);
+
+      setActiveTodo(newActiveTodo);
+      setActiveUser(user);
+    } else {
+      setActiveTodo(undefined);
+      setActiveUser(null);
+    }
+  };
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setTodos(await getTodos());
+    };
+
+    fetchTodos();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    const filteredByQuery = [...todos.filter(({ title }) => {
+      const lowerCaseTitle = title.toLocaleLowerCase();
+      const lowerCaseQuery = query.toLocaleLowerCase();
+
+      return lowerCaseTitle.includes(lowerCaseQuery);
+    }),
+    ];
```

Comment:
No need to create copy. `filter` return new array.
Why you need filter in case when query is empty?
My assumption, this filtration should be after `filterBySelect` and in case when query is not empty